### PR TITLE
Handle demangling/remangling for concurrency-related builtin types.

### PR DESCRIPTION
**Pull request by DougGregor**
Fixes rdar://75990281.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

File: stdlib/public/runtime/KnownMetadata.cpp [hunk 1]
```diff
@@ -80,6 +85,19 @@ namespace ctypes {
     /// with this, but the type isn't copyable, so most of the value
     /// operations are meaningless.
     using BB = ValueBuffer;
+
+    // Types that are defined in the _Concurrency library
+
+    // Default actor storage type.
+    struct BD {
+      void *storage[NumWords_DefaultActor];
+    };
```
Comment (rjmccall) [inline stdlib/public/runtime/KnownMetadata.cpp:94]: If these C types are expected to have the same layout characteristics as the Swift types, they need to be aligned.

Comment (DougGregor) [reply stdlib/public/runtime/KnownMetadata.cpp:94]: Good catch, thanks

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Pull request merged**